addons/draggablesprite2d/src/draggablesprite2d.gd

```diff
@@ -1,79 +1,121 @@
 @tool
-extends Area2D
+class_name DraggableSprite2D extends Area2D
+## A draggable sprite that can be dragged using the left mouse button.
 
-var sprite := Sprite2D.new()
-var collider : CollisionShape2D
-
-var is_grabbed := false :
-	set(x):
-		is_grabbed = x
-		## Emit signal if currently grabbed or not
-		if x:
-			grabbed.emit()
-		else:
-			released.emit()
-## Helps a bit to make the dragging less choppy
-var grabbed_offset := Vector2.ZERO
-## Store for the original position
-var origin := Vector2.ZERO
 
+## Emitted when the sprite is grabbed
 signal grabbed
+## Emitted when the sprite is released
 signal released
 
-## Sprite texture
-## This is a bit crap, since it can't be resized as nicely as a normal sprite2d
+
+## The input button that will be used to grab the sprite
+@export var input_method: MouseButton = MOUSE_BUTTON_LEFT
+## The texture that will be displayed by the sprite. [br]
+## Note: The texture can't be rezised as nicley as a Sprite2D.
 @export var texture : Texture2D : 
-	set(x):
-		texture = x
+	set(value):
+		texture = value
 		sprite.texture = texture
-		## Update the default collider with the shape and size of the sprite, if it exists
-		if collider and collider.shape:
-			collider.shape.size = Vector2(texture.get_width(), texture.get_height())
+		# Update the default_collider with the shape and size of the sprite, if it exists
+		update_default_collider()
 ## Whether or not the sprite should return to it's starting position when released
 @export var return_to_origin := false
 ## Whether or not it should be possible to grab the sprite
 @export var grabbable := true
 
-func _ready():
-	## Check if we need to instantiate a collider 
-	var children = get_children()
-	var needs_collider = true
-	for i in children:
-		## If a collision shape is a child, we don't need a collider
-		if i is CollisionShape2D or i is CollisionPolygon2D:
-			needs_collider = false
+
+## The Sprite node that will be used to display the texture
+var sprite := Sprite2D.new()
+## The default collider. It is automatically created and updated to match the size of the sprite
+var default_collider : CollisionShape2D
+## Whether or not the sprite is currently grabbed
+var is_grabbed := false :
+	set(value):
+		is_grabbed = value
+		if is_grabbed:
+			grabbed.emit()
+		else:
+			released.emit()
+
+
+# Helps a bit to make the dragging less choppy
+var grabbed_offset := Vector2.ZERO
+# Store for the original position
+var origin := Vector2.ZERO
+# Mouse button pressed tracker, used to essentially replicate the behavior of 'is_action_just_released'
+var mb_pressed = false
+
+
+func _ready() -> void:
+	# Connect signals
```

Comment:
In my opinion, you should avoid these small comments (L51, L56, L66, L70 (I know this one's mine, bad on my part)) where the code already explains exactly what it's doing in exactly so many terms. 

Per [Riastradh's Lisp Style Rules](https://mumble.net/~campbell/scheme/style.txt) which is obviously for Lisp but the same principle applies:
>Write comments only where the code is incapable of explaining itself.
Prefer self-explanatory code over explanatory comments.  Avoid
`literate programming' like the plague.

n.b. you shouldn't extend this logic to L6 and L8 since the documentation is surfaced in the editor.

Reply:
L51 is more of a visual seperator between code and signals, but feel free to edit them as you see fit.